pkg/ddl/partition.go

```diff
@@ -2265,13 +2266,16 @@ func (w *worker) onDropTablePartition(jobCtx *jobContext, t *meta.Mutator, job *
 		}
 
 		job.SchemaState = model.StateDeleteOnly
-		ver, err = updateVersionAndTableInfo(jobCtx, t, job, tblInfo, originalState != job.SchemaState)
+		tblInfo.Partition.DDLState = job.SchemaState
+		tblInfo.Partition.DDLAction = job.Type
+		ver, err = updateVersionAndTableInfo(jobCtx, t, job, tblInfo, true)
```

Comment:
It will always be true, since `originalState` is set to `job.SchemaState` before the switch, and is changed in each of the switch cases before the call. And it should update the Schema Version for each of the state changes.